### DISCUSSION TITLE
Bulk Request Signed URLs (instead of requesting individually for each URL)

### DISCUSSION THREAD

**Discussion by kkukshtel** [supabase/supabase#27994] [Feature Requests]
Hello! I've built a site on top of Supabase that serves images to users via private storage buckets that are exposed to end users through signed URLs. You can see a simple example here: [https://mood.site/aig5WnpL](https://mood.site/aig5WnpL).

On the backend, I have to request signed URLs for the displayed images, but given how the Storage API works I need to request each signed image individually with a dedicated storage request:

```ts
const signedUrls = await Promise.all(data.map(async (file) => {
      const ext = file.name.split('.').pop();
      const { data, error } = await supabase.storage
        .from(bucketName)
        .createSignedUrl(`${folderPath}/${file.name}`, 60 * 60, ext == "gif" ? null : {
            transform: {
              quality : 40
            }}); // URL expires in 1 hour
  
      if (error) {
        console.log(error);
        return {error: new Error('Error creating signed URL for ' + file.name + ': ' + signError.message)}
      }
      return {
        fileName: file.name,
        url : data.signedUrl,
        data : {size : dbmap.get(file.name).column_size}
      };
    }));
```

This means that every URL request is an independent API call. This is bad because:
- Storage request quota is large for a "simple" operation like just loading a page of images
- Sites with lots of images load slower primarily because they have to do additional API requests for each image

It would be nice if there was a way to just generate a list of signed URLs from an input list of paths and have that be a single API call like "Bulk Request Signed URLs". This way all the URLs could be generated in a single API call and everything could be returned a lot faster!

**Comment by kkukshtel** [comment]
Oh wow right after I posted this I found that this was already possible!

https://supabase.com/docs/reference/javascript/storage-from-createsignedurls

**Comment by nwatab** [comment]
@kkukshtel 
> Storage request quota is large for a "simple" operation like just loading a page of images
Does supabase have "Storage request quota" ?
